at_secondary/at_secondary_server/lib/src/notification/stats_notification_service.dart

```diff
@@ -39,70 +39,75 @@ class StatsNotificationService {
-  AtCommitLog? _atCommitLog;
-  DateTime _lastSentTime = DateTime.now().toUtc();
+  late String currentAtSign;
+  AtCommitLog? atCommitLog;
+  List<InboundConnection>? connectionsList;
+
+  // Counter for number of active monitor connections. Used for logging purpose.
+  int numOfMonitorConn = 0;
 
   /// Starts the [StatsNotificationService] and notifies the latest commitID
   /// to the active monitor connections.
   /// The [AtSecondaryConfig.statsNotificationJobTimeInterval] represents the time interval between the jobs.
-  void schedule() async {
+  Future<void> schedule(String currentAtSign) async {
     // If set to -1, the feature is disabled. Do nothing.
     if (AtSecondaryConfig.statsNotificationJobTimeInterval == -1) {
       _logger.info('StatsNotificationService is disabled');
       return;
     }
     _logger.info(
         'StatsNotificationService is enabled. Runs every ${AtSecondaryConfig.statsNotificationJobTimeInterval} seconds');
-    _atCommitLog =
-        await AtCommitLogManagerImpl.getInstance().getCommitLog(_currentAtSign);
-    // Setting while(true) to form an infinite loop.
-    // Runs the _schedule method as long as server is up and running.
-    while (true) {
-      await Future.delayed(
-          Duration(seconds: AtSecondaryConfig.statsNotificationJobTimeInterval),
-          _schedule);
-    }
-  }
+    this.currentAtSign = currentAtSign;
+    atCommitLog ??=
+        await AtCommitLogManagerImpl.getInstance().getCommitLog(currentAtSign);
 
-  Future<void> _schedule() async {
-    await writeStatsToMonitor();
+    // Runs the _schedule method as long as server is up and running.
+    Timer.periodic(
+        Duration(seconds: AtSecondaryConfig.statsNotificationJobTimeInterval),
+        (timer) {
+      try {
+        _logger.finer('Stats Notification Job triggered');
+        writeStatsToMonitor();
+        _logger.finer('Stats Notification Job completed');
+      } on Exception catch (exception) {
+        _logger.severe(
+            'Exception occurred when writing stats ${exception.toString()}');
+      } on Error catch (error) {
+        _logger.severe('Error occurred when writing stats ${error.toString()}');
+      }
+    });
   }
 
   /// Writes the lastCommitID to the monitor connection for every [AtSecondaryConfig.statsNotificationJobTimeInterval] seconds. Defaulted to 15 seconds
-  Future<void> writeStatsToMonitor(
-      {String? latestCommitID, String? operationType}) async {
+  void writeStatsToMonitor({String? latestCommitID, String? operationType}) {
     try {
-      latestCommitID ??= _atCommitLog!.lastCommittedSequenceNumber().toString();
+      latestCommitID ??= atCommitLog!.lastCommittedSequenceNumber().toString();
       // Gets the list of active connections.
-      var connectionsList =
-          InboundConnectionPool.getInstance().getConnections();
+      connectionsList ??= InboundConnectionPool.getInstance().getConnections();
```

Comment:
Can "connectionsList " be a local variable? The way it was before.

Reply:
Made the connectionsList global variable to pass mock connections for unit test